File: code/python/ch03/practice/practice_01_fizzbuzz.py
"""
Practice 01: FizzBuzz
==============================
Chapter 3: Decisions and Loops

PROBLEM
-------
Given a positive integer n, return the FizzBuzz sequence from 1 to n.
For each number:
  - If divisible by both 3 and 5, use "FizzBuzz"
  - If divisible by 3 only, use "Fizz"
  - If divisible by 5 only, use "Buzz"
  - Otherwise, use the number as a string

INPUT FORMAT
------------
A single line containing a positive integer n.

OUTPUT FORMAT
-------------
Print each element of the FizzBuzz sequence on its own line.

CONSTRAINTS
-----------
1 <= n <= 1000

EXAMPLES
--------
Input:  5
Output:
1
2
Fizz
4
Buzz

Input:  15
Output:
1
2
Fizz
4
Buzz
Fizz
7
8
Fizz
Buzz
11
Fizz
13
14
FizzBuzz

INSTRUCTIONS
------------
Replace the `pass` in the solve() function with your solution.
The main block at the bottom handles input/output — don't change it.
"""


def solve(n: int) -> list[str]:
    """Return the FizzBuzz sequence from 1 to n."""
    pass  # TODO: Replace this with your solution


# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    n = int(input())
    for item in solve(n):
        print(item)

File: code/cpp/ch03/practice/practice_01_fizzbuzz.cpp
/*
 * Practice 01: FizzBuzz
 * ==============================
 * Chapter 3: Decisions and Loops
 *
 * PROBLEM
 * -------
 * Given a positive integer n, return a vector of strings for numbers 1 to n:
 *   - If the number is divisible by both 3 and 5, use "FizzBuzz"
 *   - If the number is divisible by 3 only, use "Fizz"
 *   - If the number is divisible by 5 only, use "Buzz"
 *   - Otherwise, use the number as a string
 *
 * INPUT FORMAT
 * ------------
 * A single line containing a positive integer n.
 *
 * OUTPUT FORMAT
 * -------------
 * Print n lines, each being the FizzBuzz value for that number.
 *
 * CONSTRAINTS
 * -----------
 * 1 <= n <= 10000
 *
 * EXAMPLES
 * --------
 * Input:  5
 * Output:
 *   1
 *   2
 *   Fizz
 *   4
 *   Buzz
 *
 * Input:  15
 * Output:
 *   1
 *   2
 *   Fizz
 *   ...
 *   14
 *   FizzBuzz
 *
 * INSTRUCTIONS
 * ------------
 * Replace the body of solve() with your solution.
 * The main() function handles I/O -- don't change it.
 */

#include <iostream>
#include <string>
#include <vector>
using namespace std;

/**
 * Return a vector of FizzBuzz strings for numbers 1 through n.
 */
vector<string> solve(int n) {
    // TODO: Replace this with your solution
    return {};
}

// ── Do not change anything below this line ──────────────────────────
int main() {
    int n;
    cin >> n;
    vector<string> result = solve(n);
    for (const string& line : result) {
        cout << line << endl;
    }
    return 0;
}
